src/fun/anon_channel/anon.go.tmpl

```diff
@@ -0,0 +1,31 @@
+{{/*
+	Sticky message with button to submit an anonymous message.
+	See <https://yagpdb-cc.github.io/fun/anon-channel/main-cc> for more information.
+
+	Author: SoggySaussages <https://github.com/SoggySaussages>
+*/}}
+
+{{ $all_anon := true }}
+
+{{/* CONFIG ENDS */}}
+
+{{ dbSet 0 "anon_main_cc_id" .CCID }}
+
+{{ $msg := .ExecData }}
+
+{{ if and ( not $msg ) ( or $all_anon ( hasPrefix .Message.Content ( print .ServerPrefix "anon" ) ) ) }}
+	{{ deleteTrigger 0 }}
+    {{ $msg = reReplace ( printf `\A\s*\Q%s\Eanon(ymous)?\s*` .ServerPrefix ) .Message.Content "" }}
+{{ end }}
+
+{{ with $msg }}
+	{{ sendMessage nil ( cembed "author" ( sdict "name" "Anonymous" "icon_url" "https://i.imgur.com/9slnZlV.jpeg" ) "description" . ) }}
+{{ end }}
```

Comment:
At this point in your code `$msg` is guaranteed to not be nil, no?
Thus, this `with` statement is not necessary, it might even hurt readability.

Also, consider extracting the more-or-less magic string for `icon_url` to a variable; not necessarily a configuration value.

Reply:
No, it is not guaranteed to be nil if `$all_anon == false`.

Will extract though 👍